src/SplitPackages/Program.cs

```diff
@@ -0,0 +1,288 @@
+        {
+            for (int i = 0; i < source.Count; i++)
+            {
+                if (!source[i].FoundInCsv)
+                {
+                    var msg = $@"No entry in the csv file matched the following package:
+{source[i].Name}";
+                    WriteWarning(msg);
+                }
+            }
+
+            for (int i = 0; i < csv.Count; i++)
+            {
+                if (!csv[i].FoundInFolder)
+                {
+                    var msg = $@"No package found in the source folder for the following csv entry:
+{csv[i].Name}";
+                    WriteWarning(msg);
+                }
+            }
+        }
+
+        private static void EnsureDestinationDirectories(string destinationFolder, IEnumerable<string> folders)
+        {
+            foreach (var destination in folders)
+            {
+                var path = Path.Combine(destinationFolder, destination);
+                if (!Directory.Exists(path))
+                {
+                    Directory.CreateDirectory(path);
+                }
+            }
+        }
+
+        private static void SetDestinationPath(Arguments arguments, IList<PackageItem> expandedPackages)
+        {
+            foreach (var package in expandedPackages)
+            {
+                var path = Path.Combine(
+                    arguments.DestinationFolder,
+                    package.DestinationFolderName,
+                    package.Name);
+
+                package.DestinationPath = path;
+            }
+        }
+
+        private static void CopyPackages(IList<PackageItem> expandedPackages, bool whatif)
+        {
+            foreach (var package in expandedPackages)
+            {
+                WriteInformational($@"Copying package {package.Name} from
+{package.OriginPath} to
+{package.DestinationPath}");
+                if (!whatif)
+                {
+                    File.Copy(package.OriginPath, package.DestinationPath, overwrite: true);
+                }
+            }
+        }
+
+        private static void WriteInformational(string msg)
+        {
+            Console.ForegroundColor = ConsoleColor.Cyan;
+            Console.WriteLine($"[informational]: {msg}");
+            Console.ResetColor();
+        }
+
+        private static void WriteWarning(string msg)
+        {
+            Console.ForegroundColor = ConsoleColor.Yellow;
+            Console.WriteLine($"[warning]: {msg}");
+            Console.ResetColor();
+        }
+
+        private static void WriteError(string error)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($@"There was an error executing the command:
+{error}");
+            Console.ResetColor();
+            PrintUsage();
+        }
+
+        private static void PrintUsage()
+        {
+            var text = @"
+SplitPackages [help] [source packages] [csv path] [destination folder]
```

Comment:
This help doesn't match the actual usage. Instead I recommend showing the 3-4 examples of usage:

```
SplitPackages help
SplitPackages whatif X Y Z
SplitPackages X Y Z
```

And then document what each parameter uses. This avoids ambiguity in how options can be combined.